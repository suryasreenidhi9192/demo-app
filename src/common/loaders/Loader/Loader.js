import React from 'react'
import { LoaderInline } from '../../../common'
import PropTypes from 'prop-types'

import './Loader.scss'

/**
 * @public
 * Simple Loader component displayed by boolean prop
 * @param   {Boolean}     props.isLoading       Controls display of Loader
 * @param   {Node}        props.children        React Component or string to render in the body
 * @return  {Node}                              React Component
 */

const Loader = ({ isLoading, children }) => (
    isLoading &&
    <div className="loader">
        <LoaderInline message={ children } />
    </div>
) || null

Loader.propTypes = {
    /** Controls display of Loader */
    isLoading: PropTypes.bool,
    /** React Component or string to render in the body */
    children: PropTypes.node
}

export default Loader
