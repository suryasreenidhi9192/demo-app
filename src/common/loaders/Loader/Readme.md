### Loader
```jsx
    const Button = require('../../buttons/Button').default;
    const [loading, setLoading] = React.useState(false)
    const toggleState = () => {
        setLoading(true)
        setTimeout(()=>{
            setLoading(false)
        }, 5000)
    }
    const Wrapper = ({ isLoading }) => (
        <>
            <Button onClick={toggleState}>
                Show Loader for 5 Seconds
            </Button>
            <Loader isLoading={isLoading}/>
        </>
    );
    <Wrapper isLoading={loading} />
```
