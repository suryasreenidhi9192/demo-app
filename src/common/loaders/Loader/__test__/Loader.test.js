import {
    matchSnapshot
} from '../../../../../__test__/__utils__'

import Loader from '../index'

const props = {
    isLoading: true
}

const loadedProps = {
    isLoading: false
}

describe('Loader:.......................SnapShot Testing:', () => {
    test(
        'Loader Matches jest snapshot',
        matchSnapshot(Loader, props)
    )
    test(
        'Loader Doesn\'t appear if content loaded',
        matchSnapshot(Loader, loadedProps)
    )
})

