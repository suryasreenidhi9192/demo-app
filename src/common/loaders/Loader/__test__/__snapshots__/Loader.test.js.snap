// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader:.......................SnapShot Testing: Loader Doesn't appear if content loaded 1`] = `null`;

exports[`Loader:.......................SnapShot Testing: Loader Matches jest snapshot 1`] = `
<div
  className="loader"
>
  <div
    className="loader-inline"
  >
    <div
      className="loading-box"
    >
      <div
        className="spinner"
      />
      <div
        className="message"
      >
        loading...
      </div>
    </div>
  </div>
</div>
`;
